lib/commands/debug/index.js

```diff
@@ -0,0 +1,200 @@
+            cb(error);
+        }
+
+        try {
+            new ResourcesConfig(path.join(process.cwd(), CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG));
+            skillId = ResourcesConfig.getInstance().getSkillId(profile);
+        } catch (err) {
+            const error = new CliError('Failed to read project resource file. '
+                + 'Please run the command within a ask-cli project.');
+            Messenger.getInstance().error(error);
+            return cb(error);
+        }
+        if (!stringUtils.isNonBlankString(skillId)) {
+            const error = new CliError(`Failed to obtain skill-id for the given profile - ${profile} from project `
+                + `resource file ${CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG}`);
+            Messenger.getInstance().error(error);
+            return cb(error);
+        }
+
+        this._getAccessTokenForProfile(profile, cmd.debug, (tokenErr, token) => {
+            if (tokenErr) {
+                Messenger.getInstance().error(tokenErr);
+                return cb(tokenErr);
+            }
+            this._getSkillDebugflow(skillId, profile, skillCodeRegion, cmd.waitForAttach, cmd.debug, debugPort, token,
+                debugRegion, (err, debugFlow) => {
+                    if (err) {
+                        Messenger.getInstance().error(err);
+                        return cb(err);
+                    }
+                    if (cmd.waitForAttach) {
+                        Messenger.getInstance().info(`Debugging session will wait until inspector is attached at port - ${debugPort}`);
+                    }
+                    debugFlow.execCommand();
+                });
+        });
+    }
+
+    _getAccessTokenForProfile(profile, debug, callback) {
+        const authorizationController = new AuthorizationController({
+            auth_client_type: 'LWA',
+            doDebug: debug
+        });
+        authorizationController.tokenRefreshAndRead(profile, (tokenErr, token) => {
+            if (tokenErr) {
+                return callback(tokenErr);
+            }
+            callback(null, token);
+        });
+    }
+
+    _getHostedSkillRuntime(smapiClient, skillId, callback) {
+        smapiClient.skill.alexaHosted.getAlexaHostedSkillMetadata(skillId, (err, response) => {
+            if (err) {
+                return callback(err);
+            }
+            if (response.statusCode >= 300) {
+                const error = jsonView.toString(response.body);
+                return callback(error);
+            }
+            const { runtime } = response.body.alexaHosted;
+            if (!stringUtils.isNonBlankString(runtime)) {
+                return callback(`Unable to determine runtime of the hosted skill - ${skillId}`);
+            }
+            callback(null, runtime);
+        });
+    }
+
+    _getSkillDebugflow(skillId, profile, skillCodeRegion, waitForAttach, debug, debugPort, token, debugRegion, callback) {
```

Comment:
i really think this getDebugFlow method shouldbe in the helper. Or at least move AHS NHS's getDebugFlow into helper, so the if-else here won't be a large code block. You will see how convenient it can be when you write unit test :)

Reply:
I agree!